core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -195,14 +317,23 @@ object ConfigCommand {
       CommandLineUtils.checkInvalidArgs(parser, options, alterOpt, Set(describeOpt))
       CommandLineUtils.checkInvalidArgs(parser, options, describeOpt, Set(alterOpt, addConfig, deleteConfig))
       if(options.has(alterOpt)) {
-        require(options.has(entityName), "--entity-name must be specified with --alter")
+        if(!options.has(entityName) && !options.has(entityDefault))
```

Comment:
entityDefault can only be specified for client/user for now. So we probably need to adjust the check a bit.